### PR TITLE
docs: add detailed instructions for setting up external root CA

### PR DESCRIPTION
**Pull request by dnephin**
Branched from #11910, this PR adds detailed instructions for how to use the new feature.

### CODE REVIEW

**Review by karl-cardenas-coding — CHANGES_REQUESTED** [review]
@trujillo-adam  this looks good to me but let's tag @webdog and @danielehc to review this as they recently interacted heavily with this doc page.

**Review by danielehc — APPROVED** [review]
LGTM. I like the tables to show the different options and the explanation on the different necessity of each field.

File: website/content/docs/connect/ca/vault.mdx [hunk 1]
```diff
@@ -79,122 +79,115 @@ connect {
...
+| `TLSServerName`, <br/>`tls_server_name` | Specifies the SNI host when connecting to Vault via TLS. | String | Optional |
+| `TLSSkipVerify`, <br/>`tls_skip_verify` | Enables SSL peer validation enforcement. Default is `false`. | Boolean | Optional |
+| `Namespace`, <br/>`namespace` | Specifies the Vault namespace that the `Token` and PKI certificates are a part of. Requries Vault Enterprise (see [Namespaces](https://www.vaultproject.io/docs/enterprise/namespaces) for additional information). | String | Optional | 
+
+### Auth Methods
+
+Auth methods are components in Vault that perform authentication. They are responsible for assigning identity and a set of policies to a user. See [Vault Auth Methods](https://www.vaultproject.io/docs/auth) for information about configuring individual auth methods in Vault.
+
+If auth method is provided, Consul will obtain a new token from Vault when the token can no longer be renewed. 
+Specify following parameters to define an `AuthMethod` / `auth_method` object.
```
Comment (webdog) [inline website/content/docs/connect/ca/vault.mdx:104]:
```suggestion
Specify the following parameters to define an `AuthMethod` / `auth_method` object.
```

File: website/content/docs/connect/ca/vault.mdx [hunk 2]
```diff
@@ -79,122 +79,115 @@ connect {
...
-  !> **Warning:** You must either provide a token or configure an auth method below.
+### Root PKI Paths
 
-- `AuthMethod` / `auth_method` (`map: nil`) - Vault auth method to use for logging in to Vault.
-  Please see [Vault Auth Methods](https://www.vaultproject.io/docs/auth) for more information
-  on how to configure individual auth methods. If auth method is provided, Consul will obtain a
-  a new token from Vault when the token can no longer be renewed.
+The `RootPKIPath`, `root_pki_path` parameter specifies the PKI path to the root certificate. 
+Consul uses the root certificate to sign an intermediate certificate for each Consul datacenter, including the primary datacenter. 
+When WAN Federation is enabled, each secondary datacenter must use the same Vault cluster and share the same `root_pki_path` with the primary datacenter.
```
Comment (webdog) [inline website/content/docs/connect/ca/vault.mdx:122]:
```suggestion
When WAN Federation is enabled, ensuing datacenters must be configured to use the same Vault cluster and share the same `root_pki_path` as the primary datacenter.
```

File: website/content/docs/connect/ca/vault.mdx [hunk 3]
```diff
@@ -79,122 +79,115 @@ connect {
...
 
-   - `Type`/ `type` (`string: ""`) - The type of Vault auth method.
+If the path does not exist, Consul will mount a new PKI secrets engine at the specified path with the `RootCertTTL` value as the root certificate's TTL. 
+Consul never modifies data within the root PKI path after the path has been mounted and initialized.
```
Comment (webdog) [inline website/content/docs/connect/ca/vault.mdx:125]:
```suggestion
Consul does not modify data within the root PKI path after the path has been mounted and initialized.
```

File: website/content/docs/connect/ca/vault.mdx [hunk 4]
```diff
@@ -79,122 +79,115 @@ connect {
...
 
-  To use an intermediate certificate as the primary CA in Consul initialize the
-  `RootPKIPath` in Vault with a PEM bundle. The first certificate in the bundle
-  must be the intermediate certificate that Consul will use as the primary CA.
-  The last certificate in the bundle must be a root certificate. The bundle
-  must contain a valid chain, where each certificate is followed by the certificate
-  that authorized it.
+The [`IntermediatePKIPath` (`intermediate_pki_path`)](#intermediatepkipath) is
+the PKI path used for storing the intermediate CA that was signed by the root certificate.
+The intermediate is used to sign all leaf certificates.
```
Comment (webdog) [inline website/content/docs/connect/ca/vault.mdx:143]:
```suggestion
The intermediate certificate is used to sign all subsequent leaf certificates.
```

File: website/content/docs/connect/ca/vault.mdx [hunk 5]
```diff
@@ -79,122 +79,115 @@ connect {
...
+| `CAPath`, <br/>`ca_path` | Specifies a path to a folder containing CA certificates to be used for Vault communication. If unspecified, the default system CA bundle is used (refer to the documentation for your OS and version). | String | Optional |
+| `CertFile`, <br/>`cert_file` | Specifies the path to the certificate used for Vault communication. | String | Required when `KeyFile`, `key_file` is configured  |
+| `KeyFile`, <br/>`key_file` | Specifies the path to the private key used for Vault communication. | String | Required when `CertFile`, `cert_file` is configured. |
+| `TLSServerName`, <br/>`tls_server_name` | Specifies the SNI host when connecting to Vault via TLS. | String | Optional |
+| `TLSSkipVerify`, <br/>`tls_skip_verify` | Enables SSL peer validation enforcement. Default is `false`. | Boolean | Optional |
+| `Namespace`, <br/>`namespace` | Specifies the Vault namespace that the `Token` and PKI certificates are a part of. Requries Vault Enterprise (see [Namespaces](https://www.vaultproject.io/docs/enterprise/namespaces) for additional information). | String | Optional | 
+
```
[the remark this answers is not in the document]
Comment (trujillo-adam) [reply website/content/docs/connect/ca/vault.mdx:98]: Yeah, the construct you called out was what I was trying to get away from. I can see how the term "API Key" could be confusing. I would still like to keep them in the same column to reinforce the idea that the keywords perform the same function. How about "Parameter (API, Agent)"? Or just "API, Agent"? "Resource, Agent"?  I have such a love/hate relationship with md tables.

**Comment by kisunji** [comment]
Just FYI @webdog & @trujillo-adam, @dnephin has recently left Hashicorp so this PR may need some additional love to bring across the finish line.

I can get to it in a week or so but lack the background knowledge for this feature set so this might need some team involvement

**Comment by trujillo-adam** [comment]
I applied as much of the feedback and answered as many questions as I could. Most of @webdog 's question will need to be addressed by @kisunji when he has a chance next week. The remaining questions have to do with Nathan's original additions, which, unfortunately, we can't ask him to clarify. 

I updated the table column headers in a way that I hope addresses @webdog 's concerns.

On hold for now.

File: website/content/docs/connect/ca/vault.mdx (hunk 5 above)
[the remark this answers is not in the document]
Comment (blake) [reply website/content/docs/connect/ca/vault.mdx]: The API takes a JSON object as input. In JSON, the keys of an object are referred to as a field.

Would it make sense to break these into two separate columns with the following headings?

| Configuration parameter | API field | Description | Type | Required |
| --- | --- | --- | --- | --- |
| `Address` |  `address` | Specifies the address of the Vault server. | String | Yes |

File: website/content/docs/connect/ca/vault.mdx [hunk 6]
```diff
@@ -79,122 +79,115 @@ connect {
...
+Consul will periodically generate new intermediates before the existing certificate
+expires. Therefore, Consul requires write access to this path. This path is managed
+entirely by Consul, it should not be mounted or initialized.
 
-- `IntermediatePKIPath` / `intermediate_pki_path` (`string: <required>`) -
-  The path to a PKI secrets engine for the generated intermediate certificate.
-  This certificate will be signed by the configured root PKI path. If this
-  path does not exist, Consul will attempt to mount and configure this
-  automatically. 
+If the intermediate PKI path does not exist, Consul will attempt to mount and initialize it. This requires additional privileges by the Vault token.
```
[the remark this answers is not in the document]
Comment (blake) [reply website/content/docs/connect/ca/vault.mdx:148]:
```suggestion
If the intermediate PKI path does not exist, Consul will attempt to mount and initialize it. This requires [additional privileges](#consul-managed-pki-paths) by the Vault token.
```

File: website/content/docs/connect/ca/vault.mdx [hunk 7]
```diff
@@ -79,122 +79,115 @@ connect {
...
-  to the default system CA bundle, which varies by OS and version.
+To use an external root CA with the Vault provider, you must initialize the
+`RootPKIPath` before configuring Consul. The following steps describe how to initialize the path:
 
-- `CAPath` / `ca_path` (`string: ""`) - Specifies an optional path to a folder
-  containing CA certificates to be used for Vault communication. If
-  unspecified, this will fallback to the default system CA bundle, which
-  varies by OS and version.
+1. Enable the PKI secrets engine at the path (`connect_root`) that will be used as
+   `RootPKIPath`. Optionally you may also tune the path using `vault secrets tune`.
```
[the remark this answers is not in the document]
Comment (blake) [reply website/content/docs/connect/ca/vault.mdx:156]:
```suggestion
1. Enable the PKI secrets engine at the path (`connect_root`) that will be used as
   `RootPKIPath` in the Connect CA configuration. Optionally you may also tune the path using `vault secrets tune`.
```

[the remark this answers is not in the document]
Comment (blake) [reply website/content/docs/connect/ca/vault.mdx:156]: Tune apparently allows customizing details of the PKI path. https://www.vaultproject.io/docs/commands/secrets/tune

File: website/content/docs/connect/ca/vault.mdx [hunk 8]
```diff
@@ -79,122 +79,115 @@ connect {
...
-  string used to set the SNI host when connecting to Vault via TLS.
+   ```shell-session
+   $ vault write connect_root/intermediate/generate/internal
+   ```
 
-- `TLSSkipVerify` / `tls_skip_verify` (`bool: false`) - Specifies if SSL peer
-  validation should be enforced.
+1. Use the external root CA to sign the CSR. The exact steps depend on where the root 
+CA is stored.  If you are using vault to store the root CA use
+[sign-intermediate](https://www.vaultproject.io/api-docs/secret/pki#sign-intermediate).
```
[the remark this answers is not in the document]
Comment (blake) [reply website/content/docs/connect/ca/vault.mdx:170]: This enhancement assumes that the external root CA is stored in a location that is completely external to Consul or Vault, such as protected in an HSM. The specifics of signing keys from this system will vary based on the customer's environment.

File: website/content/docs/connect/ca/vault.mdx [hunk 9]
```diff
@@ -79,122 +79,115 @@ connect {
...
+   signed it.
 
-@include 'http_api_connect_ca_common_options.mdx'
+   ```shell-session
+   $ cat signed-intermediate.pem root-ca.pem > bundle.pem
+   ```
 
-## Root and Intermediate PKI Paths
+1. Write the bundle.pem to the `RootPKIPath` with
+[`set-signed`](https://www.vaultproject.io/api-docs/secret/pki#set-signed-intermediate).
```
[the remark this answers is not in the document]
Comment (blake) [reply website/content/docs/connect/ca/vault.mdx:180]:
```suggestion
1. Write the `bundle.pem` to the `RootPKIPath` with
[`set-signed`](https://www.vaultproject.io/api-docs/secret/pki#set-signed-intermediate).

```shell-session
vault write connect_root/intermediate/set-signed certificate=@bundle.pem
```
```

**Pull request closed without merging**